gms/factories/src/main/java/com/linkedin/gms/factory/common/GraphServiceFactory.java

```diff
@@ -2,27 +2,56 @@
 
 import com.linkedin.metadata.graph.GraphService;
 import com.linkedin.metadata.graph.Neo4jGraphService;
+import com.linkedin.metadata.graph.elastic.ElasticSearchGraphService;
+import com.linkedin.metadata.utils.elasticsearch.IndexConvention;
 import javax.annotation.Nonnull;
+import org.elasticsearch.client.RestHighLevelClient;
 import org.neo4j.driver.Driver;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.DependsOn;
 import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
 
 
 @Configuration
-@Import({Neo4jDriverFactory.class})
+@Import({Neo4jDriverFactory.class, ElasticSearchGraphServiceFactory.class})
 public class GraphServiceFactory {
+  @Autowired
+  @Qualifier("elasticSearchGraphService")
+  private ElasticSearchGraphService _elasticSearchGraphService;
+
   @Autowired
   @Qualifier("neo4jDriver")
   private Driver neo4jDriver;
 
+  @Autowired
+  @Qualifier("elasticSearchRestHighLevelClient")
+  private RestHighLevelClient searchClient;
+
+  @Autowired
+  @Qualifier(IndexConventionFactory.INDEX_CONVENTION_BEAN)
+  private IndexConvention indexConvention;
+
+  @Value("${GRAPH_SERVICE:neo4j}")
+  private String graphService;
+
   @Nonnull
-  @DependsOn({"neo4jDriver"})
+  @DependsOn({"neo4jDriver", "elasticSearchGraphService"})
   @Bean(name = "graphService")
+  @Primary
   protected GraphService createInstance() {
-    return new Neo4jGraphService(neo4jDriver);
+    if (graphService.equalsIgnoreCase("neo4j")) {
```

Comment:
In the future we should be able to specify the fully qualified class name in the config and then just instiantiate it with some context..

eg. graphServiceImpl: "com.linkedin.metadata.graph.elastic.ElasticSearchGraphService"